ow_materials/include/AxisAlignedGrid.h

```diff
@@ -0,0 +1,60 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef AXIS_ALIGNED_GRID_H
+#define AXIS_ALIGNED_GRID_H
+
+#include <vector>
+#include <memory>
+
+#include <ignition/math/AxisAlignedBox.hh>
+
+namespace ow_materials {
+
+template <typename T>
+class AxisAlignedGrid {
+public:
+  AxisAlignedGrid(double const x0, double const y0, double const z0,
+                  double const x1, double const y1, double const z1,
+                  double const side_length, T const initial_value);
+  ~AxisAlignedGrid() = default;
+
+  AxisAlignedGrid() = delete;
+  AxisAlignedGrid(const AxisAlignedGrid&) = delete;
+  AxisAlignedGrid& operator=(const AxisAlignedGrid&) = delete;
+
+  const ignition::math::Vector3d &getDiagonal() const;
+  const ignition::math::Vector3d &getMaxCorner() const;
+  const ignition::math::Vector3d &getMinCorner() const;
+  const ignition::math::Vector3d &getCenter() const;
+
+  inline T &getCellValue(size_t const i, size_t const j, size_t const k) {
```

Comment:
The general consensus I read is to `const` anything and everything that is truly constant. I can understand how it might seem excessive here, especially for a single line inlined function, but using `const` can also assist the compiler in making optimizations. This is also one of those low-level methods that will be called often and more than any other method defined in this PR, so I would really prefer to be as rigorous in its definition as possible.

Reply:
I strongly disagree with said consensus, in favor of code compactness and readability, but I don't want to argue this point.  Note that since this is a trivial inline function, I can't see any meaningful compiler optimization here.

Reply:
My main view on this matter is that arguments passed by value are _copies_, and I see no reason to force immutability on copies, and enforce an implementation detail that is invisible to callers and in principle (to me) no one should even care about.

Reply:
I have removed `const` from inline function parameters because I will concede the benefit there is minimal if not zilch to the best of my knowledge. I have also removed `const` from the prototype of `AxisAlignedGrid`'s constructor, but kept it in the implementation of the function. This way users don't have to see `const` cluttering up the header file, but the actual implementation still has the benefit of it. 

I stick to the stance that it is _always_ better to have the compiler throw an error when a variable you expect to to never mutate is being mutated. There are many reasons this may happen ranging from a typo to a bug in an API. The larger the scope of a variable, the more possibility of unwarranted mutation, and therefore the bigger the benefit of making the variable `const`.